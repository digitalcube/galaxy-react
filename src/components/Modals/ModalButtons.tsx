import React, { FC, PropsWithChildren } from 'react';
import { useModal } from './ModalContext';

export type ModalButtonsProps = PropsWithChildren<{
  cancelButtonLabel?: string;
  showCancel?: boolean;
  className?: string;
}>;
export const ModalButtons: FC<ModalButtonsProps> = ({
  children,
  cancelButtonLabel = 'Cancel',
  showCancel = true,
  className,
}) => {
  const { isOpen, dismiss } = useModal();
  return (
    <div className={`mt-4 modal-buttons${className ? ` ${className}` : ''}`}>
      {children}
      {showCancel && dismiss ? (
        <button
          className="mt-3 font-weight-bold btn btn-block btn-bordered border-0"
          data-dismiss="modal"
          onClick={() => isOpen(false)}
        >
          {cancelButtonLabel}
        </button>
      ) : null}
    </div>
  );
};
