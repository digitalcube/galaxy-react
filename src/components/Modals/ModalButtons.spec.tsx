import { render } from '@testing-library/react';
import React from 'react';
import { Default as ModalButtons } from './ModalButtons.stories';

describe('ModalButtons', () => {
  it('to match snapshot [default]', () => {
    const { asFragment } = render(
      <ModalButtons {...(ModalButtons.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('to match snapshot [hide cancel button]', () => {
    const { asFragment } = render(<ModalButtons showCancel={false} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
