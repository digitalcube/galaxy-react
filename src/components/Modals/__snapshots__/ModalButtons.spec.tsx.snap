// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalButtons to match snapshot [default] 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-labelledby="modal-label"
    aria-modal="true"
    class="modal show d-block"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content "
      >
        <div
          class="modal-body px-4 py-4"
        >
          <div
            class="mt-4 modal-buttons"
          >
            <button
              class="mt-3 font-weight-bold btn btn-block btn-bordered border-0"
              data-dismiss="modal"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="modal-backdrop show"
  />
</DocumentFragment>
`;

exports[`ModalButtons to match snapshot [hide cancel button] 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-labelledby="modal-label"
    aria-modal="true"
    class="modal show d-block"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content "
      >
        <div
          class="modal-body px-4 py-4"
        >
          <div
            class="mt-4 modal-buttons"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="modal-backdrop show"
  />
</DocumentFragment>
`;
